### PR TITLE
[1.6.0] Security Section

### LINKED ISSUE #225: Add security section
Need a security page:

Here is a good example of a threat model: https://sse-secure-systems.github.io/connaisseur/v2.0.0/threat_model/

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
Signed-off-by: ShubhamPalriwala <spalriwalau@gmail.com>

Closes #225 

@JimBugwadia Have drafted this based on our [google doc ](https://docs.google.com/document/d/1tFEo2qhsmWcjkRUu3QxYYYbr0BQluskS_0iEV_GspgA/edit#heading=h.481dqd5dgjxh). We can continue making changes here now.

### CODE REVIEW

File: content/en/security/_index.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+Kyverno is available via the [ArtifactHub page](https://artifacthub.io/packages/helm/kyverno/kyverno) also. One can also look at the auto-generated [Security Report](https://artifacthub.io/packages/helm/kyverno/kyverno?modal=security-report) generated by ArtifactHub for all the releases.
+
+## RBAC Configuration for Kyverno
+
+Kyverno installs with the roles defined [here](https://github.com/kyverno/kyverno/blob/main/definitions/k8s-resource/clusterroles.yaml)
+These roles have been taken from [our self-assessment document](https://github.com/kyverno/tag-security/blob/main/assessments/projects/kyverno/self-assessment.md#kyverno-roles).
+
+- kyverno:admin-policies
+
+```
```
Comment (chipzoller) [inline content/en/security/_index.md:99]: Please try to use languages for all fenced code blocks.

**Comment by JimBugwadia** [comment]
@ShubhamPalriwala @chipzoller - I moved the security section to:

https://deploy-preview-400--kyverno.netlify.app/docs/security/

The roles section is dependent on this PR: https://github.com/kyverno/kyverno/pull/2799

Please review, so we can wrap this up!

**Review by JimBugwadia — APPROVED** [review]

**Comment by ShubhamPalriwala** [comment]
Hey @JimBugwadia, this is very well crafted!
Looks great to me just a couple of small suggestions:
- Can we just have the command inside the backticks so that the users can directly copy it and use it?
image

- Can we now remove the signing and SBOM from the Installation Page [here](https://deploy-preview-400--kyverno.netlify.app/docs/installation/#verifying-kyverno-image-signatures-using-cosign) as it seems redundant!

If you want. I can make these small changes and commit them

**Comment by JimBugwadia** [comment]
@ShubhamPalriwala - sounds good. Please proceed with the proposed changes.

**Comment by ShubhamPalriwala** [comment]
Done! @JimBugwadia Everything looks good to me now

File: content/en/docs/security/_index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+weight: 68
+---
+
+Kyverno serves an admission controller and is a critical component of the Kubernetes control-plane. It is important to properly secure and monitor Kyverno. This section provides guidance on securing Kyverno and the security processes for the Kyverno project.
+
+## Disclosure Process
+
+Security vulnerabilities are best handled swiftly and discretely with the goal of minimizing the total time users remain vulnerable to exploits.
+
+If find or suspect a vulnerability please send an email to the security group at kyverno-security@googlegroups.com with the following information:
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:13]: "If you find or suspect a vulnerability, please email the security group at..."

File: content/en/docs/security/_index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+- description of the problem
+- precise and detailed steps (include screenshots) that created the problem
+- the affected version(s)
+- any known mitigations
+
+Kyverno security response team will send an initial acknowledgement of the disclosure in 3-5 working days. Once the vulnerability and mitigation are confirmed, the team will plan to release any necessary changes based on the severity and complexity. Additional details on the security policy and processes are available in the Kyverno [git repo](https://github.com/kyverno/kyverno/blob/main/SECURITY.md).
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:19]: "The Kyverno security response team..."

File: content/en/docs/security/_index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+## Issues
+
+All security related issues are labeled as `security` and can be viewed with this query:
+
+  [https://github.com/kyverno/kyverno/labels/security](https://github.com/kyverno/kyverno/labels/security)
+
+
+## Release Artifacts
+
+With every release, Kyverno uploads the following artifacts:
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:35]: Although they aren't on the releases page, we also upload the container images stated here: https://github.com/orgs/kyverno/packages?repo_name=kyverno

File: content/en/docs/security/_index.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+- kyverno-cli_v<version_number>_darwin_x86_64.tar.gz
+- kyverno-cli_v<version_number>_linux_x86_64.tar.gz
+- kyverno-cli_v<version_number>_windows_x86_64.zip
+- Source code (zip)
+- Source code (tar.gz)
+
+
+## Verifying Kyverno Images
+
+Kyverno container images are signed using Cosign. To verify the container image, download the organization public key (https://github.com/kyverno/kyverno/blob/main/cosign.pub) into a file named cosign.pub and then:
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:46]: Link to Cosign project.

"file named `cosign.pub`" (files as code formatted)

File: content/en/docs/security/_index.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+```sh
+Verification for kyverno/kyverno:latest --
+The following checks were performed on each of these signatures:
+  - The cosign claims were validated
+  - The signatures were verified against the specified public key
+  - Any certificates were verified against the Fulcio roots.
+[{"critical":{"identity":{"docker-reference":"ghcr.io/kyverno/kyverno"},"image":{"docker-manifest-digest":"sha256:a847df12e2c1cab19af9d1bb34e599cb56cf57639c7d5c958a4bb568c1dad8f6"},"type":"cosign container image signature"},"optional":null}]
+```
+
+All 3 of Kyverno images can be verified: `kyvernopre`, `kyverno`, and `kyverno-cli`.
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:72]: "All three Kyverno images, <insert image names here>, can be verified."

File: content/en/docs/security/_index.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+## Fetching the SBOM for Kyverno
+
+An SBOM (Software Bill of Materials) in CycloneDX JSON format is published for each Kyverno release. To download the SBOM for a specific version, install cosign and run:
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:76]: "Cosign" (proper noun)

File: content/en/docs/security/_index.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+Kyverno uses [Scorecards by OSSF](https://github.com/ossf/scorecard) to maintain repository-wide security standards. The current OSSF/scorecard score for Kyverno can be found in this [tracker issue](https://github.com/kyverno/kyverno/issues/2617). The Kyverno team is committed to achieving and maintaining a high score. Contributions are welcome.
+
+## Vulnerability Scan Reports
+
+The Kyverno Helm Chart is available via the [ArtifactHub page](https://artifacthub.io/packages/helm/kyverno/kyverno) along with an auto-generated [Security Report](https://artifacthub.io/packages/helm/kyverno/kyverno?modal=security-report) generated by ArtifactHub for all the releases.
+
+
+## Threat Model
+
+The Kubernetes SIG Security team has defined an [Admission Control Threat Model](https://docs.google.com/document/d/1tQTZ6wrD2udbUGL7xhc3SRxqyTvS8FQUlL59KHYHcY8/edit#heading=h.bx94h7ux7k41). It is highly recommended to understand the identified threats and available mitigations.
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:99]: Link to Kubernetes SIG Security would be nice.

File: content/en/docs/security/_index.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+The following sections discuss related best practices for Kyverno:
+
+### Pod security
+
+The following pod security best practices are followed for all Kyverno containers:
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:105]: "Pod"

File: content/en/docs/security/_index.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+- `kyverno-resource-mutating-webhook-cfg`: handles resource admission requests to apply matching Kyverno mutate policy rules.
+- `kyverno-verify-mutating-webhook-cfg`: periodically tests Kyverno webhook configurations 
+
+Kyverno creates the following validating webhook configurations:
+- `kyverno-policy-validating-webhook-cfg`: validates Kyverno policies with checks that cannot be performed via schema validation
+- `kyverno-resource-validating-webhook-cfg`: handles resource resource admission requests to apply matching Kyverno validate policy rules.
+
+#### Webhook Failure Mode
+
+Kyverno policies are configured to `fail-closed` by default. This setting can be tuned on a [per policy basis](/docs/writing-policies/policy-settings/). Kyverno uses the configured policy set to automatically configure webhooks.
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:143]: `fail-closed` here isn't code and should be excluded from code formatting.

File: content/en/docs/security/_index.md [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+
+#### Webhook authentication and encryption
+
+By default, Kyverno automatically generates and manage TLS certificates used for mutual authentication with the API server, and encryption of network traffic. To customize, please refer to the details in the [installation section](/docs/installation/#certificate-management).
+
+### Recommended policies
+
+The Kyverno community manages a set of [sample policies](/policies/).
+
+At a minimum, the [pod security](/policies/pod-security/) and [best practices](/policies/?policytypes=Best%2520Practices) policy sets are recommended for use.
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:153]: The Pod Security Standard

File: content/en/docs/security/_index.md [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+### Securing policies
+
+Kyverno policies can be used to mutate and generate configurations, and access to policies should be protected using RBAC.
```
Comment (chipzoller) [inline content/en/docs/security/_index.md:157]: "...and generate resources..."

**Review by chipzoller — CHANGES_REQUESTED** [review]
Please see in-line comments.

**Comment by chipzoller** [comment]
Hold for 1.6.0

**Comment by JimBugwadia** [comment]
Thanks @chipzoller! I've addressed the comments.

**Review by chipzoller — APPROVED** [review]

**Review by JimBugwadia — APPROVED** [review]

**Pull request merged**